pkg/cloud/cloud.go

```diff
@@ -18,6 +18,40 @@ import (
 	"github.com/openshift/cluster-cloud-controller-manager-operator/pkg/cloud/vsphere"
 )
 
+// cloudConfigTransformer function transforms the source config map using the input infrastructure.config.openshift.io object.
+// only the data and binaryData field of the output ConfigMap will be respected by consumer of the transformer.
+type cloudConfigTransformer func(source string, infra *configv1.Infrastructure) (string, error)
+
+// GetCloudConfigTransformer returns the function that should be used to transform
+// the cloud configuration config map
+func GetCloudConfigTransformer(platformStatus *configv1.PlatformStatus) (cloudConfigTransformer, error) {
+	switch platformStatus.Type {
+	case configv1.AlibabaCloudPlatformType:
+		return common.NoOpTransformer, nil
+	case configv1.AWSPlatformType:
+		// FIXME: We need to implement a transformer for this. Currently we're
+		// relying on CCO to do the heavy lifting for us.
+		return nil, nil
+	case configv1.AzurePlatformType:
+		// FIXME: We need to implement a transformer for this. Currently we're
+		// relying on CCO to do the heavy lifting for us.
+		return nil, nil
```

Comment:
Should these not be returning the `NoOpTransformer` for now? What happens when this returns a nil, is the caller handling that?

Reply:
@stephenfin can we return NoOpTransformer there and remain logic around syncing managed config untouched? Or its necessary to apply transformations on unmanaged config?

Reply:
I intentionally wanted to treat these as "special" because they're the only things I can see that *definitely* need to refer to the CCO-generated configmaps. If I didn't have this distinction, I'd have to either (a) pull in CCO's configmap for everything or (b) pull it in for everything *except* the clouds that I have a non-noop transformer for (currently just OpenStack) [*] Both of these will ultimately result in CCO being involved in this loop for way longer than necessary so this helps minimize larger tech debt at the cost of a small bit of (temporary) complexity.

[*] There's also (c), which is a variant of (b): add special transformers for the other clouds != Azure or AWS (or OpenStack). This adds the downside of introducing code duplication since we'd be effectively duplicating the `NoOpTransformer` for these other clouds. It's also kind of a variation of what we're currently doing.

Reply:
I've added more comments here to make this a small bit clearer